plugins/by-name/image/default.nix

```diff
@@ -45,143 +47,97 @@ in
       Auto show/hide images when the editor gains/looses focus.
     '';
 
-    tmuxShowOnlyInActiveWindow = helpers.defaultNullOpts.mkBool false ''
+    tmux_show_only_in_active_window = defaultNullOpts.mkBool false ''
       Auto show/hide images in the correct Tmux window (needs visual-activity off).
     '';
 
-    hijackFilePatterns =
-      helpers.defaultNullOpts.mkListOf types.str
+    hijack_file_patterns =
+      defaultNullOpts.mkListOf types.str
         [
           "*.png"
           "*.jpg"
           "*.jpeg"
           "*.gif"
           "*.webp"
+          "*.avif"
         ]
         ''
           Render image files as images when opened.
         '';
   };
 
-  config = mkIf cfg.enable {
-    extraPlugins = [ cfg.package ];
-    extraLuaPackages = ps: [ ps.magick ];
+  settingsExample = {
+    backend = "kitty";
+    max_width = 100;
+    max_height = 12;
+    max_height_window_percentage.__raw = "math.huge";
+    max_width_window_percentage.__raw = "math.huge";
+    window_overlap_clear_enabled = true;
+    window_overlap_clear_ft_ignore = [
+      "cmp_menu"
+      "cmp_docs"
+      ""
+    ];
+  };
 
+  extraConfig = cfg: {
     extraPackages = [
       # In theory, we could remove that if the user explicitly disables `downloadRemoteImages` for
       # all integrations but shipping `curl` is not too heavy.
       cfg.curlPackage
-    ] ++ optional (cfg.backend == "ueberzug") cfg.ueberzugPackage;
-
-    extraConfigLua =
-      let
-        setupOptions =
-          with cfg;
-          {
-            inherit backend;
-            integrations =
-              let
-                processIntegrationOptions = v: {
-                  inherit (v) enabled;
-                  clear_in_insert_mode = v.clearInInsertMode;
-                  download_remote_images = v.downloadRemoteImages;
-                  only_render_image_at_cursor = v.onlyRenderImageAtCursor;
-                  inherit (v) filetypes;
-                  resolve_image_path = v.resolveImagePath;
-                };
-              in
-              mapAttrs (_: processIntegrationOptions) integrations;
-            max_width = maxWidth;
-            max_height = maxHeight;
-            max_width_window_percentage = maxWidthWindowPercentage;
-            max_height_window_percentage = maxHeightWindowPercentage;
-            window_overlap_clear_enabled = windowOverlapClearEnabled;
-            window_overlap_clear_ft_ignore = windowOverlapClearFtIgnore;
-            editor_only_render_when_focused = editorOnlyRenderWhenFocused;
-            tmux_show_only_in_active_window = tmuxShowOnlyInActiveWindow;
-            hijack_file_patterns = hijackFilePatterns;
-          }
-          // cfg.extraOptions;
-      in
-      ''
-        require('image').setup(${lib.nixvim.toLuaObject setupOptions})
-      '';
+
+    ] ++ lib.optional (cfg.settings.backend == "ueberzug") cfg.ueberzugPackage;
+
+    extraLuaPackages = ps: [ ps.magick ];
```

Comment:
This feels like it should be conditional on some option?

Reply:
Looks like it listed as required in the docs for NixOS. Wondering if we should upstream that.

Reply:
That's reasonable. I could hypothetically imagine someone wanting to use a different `magick` package, e.g. with some overrides or a different version. Or maybe someone wanting it to be found impurely from the host system.

But this is somewhat contrived so I'm fine with not supporting that usage.

Reply:
This looks like the magick lua package though, right? I think it's different than an overrideable imagemagick binary.

Reply:
Looking upstream, we already pass this in the lua derivation.

Reply:
Yeah, if it breaks it's an issue with how we fetch/wrap dependencies.

Reply:
I wonder if this is another case that our combinedPlugins impl can't currently handle :grin:

Reply:
> I wonder if this is another case that our combinedPlugins impl can't currently handle 😁

*PTSD triggers*